### ISSUE TITLE
Spawned sub-tasks do not log to forest.

### ISSUE THREAD

**Issue by Firstyear** [QnnOkabayashi/tracing-forest#3]
```
     Running `target/debug/forest_bug`
00000000-0000-0000-0000-000000000000 2022-01-23T03:36:25.359390322+00:00 ERROR    🚨 [error]: Waiting on signal
^C00000000-0000-0000-0000-000000000000 2022-01-23T03:36:26.862968017+00:00 ERROR    🚨 [error]: Stopping
[william@pyrite 13:36] 0 forest_bug/ # cat src/main.rs
use tokio::signal;

#[tracing_forest::main]
#[tokio::main]
async fn main() {
    tokio::task::spawn(async {
        tracing::error!("Test message");
    });

    tracing::error!("Waiting on signal");
    let _ = signal::ctrl_c().await;
    tracing::error!("Stopping");
}
```

**Comment by Firstyear** [comment]
Okay, the issue is that on_future only uses set_default, not set_global_default. This means any other tasks that are spawned don't attach to this subscriber. 

I think that this will require a way to "build" the subscriber, and a way to signal to it that shutdown is now occurring, so that the subscriber can be set to the global default, and during program stop, the shutdown can be flagged so that any remaining messages can be flushed and then the queue closed. 

Additionally, you likely will need some logic for the tx side of the queue so that if anything is logged posted writer stop they are set to immediate instead.

**Comment by Firstyear** [comment]
@QnnOkabayashi I'm happy to work on this, but I'd like your advice about it :) 

Im thinking that there is a new "on_default_future" rather than "on_future" which sets the default tracing subscriber to forest. To track if everything is shutdown, We can't use the drop guard in the same way, since there "may" but spawned tasks that persist, so I think that we have to do a oneshot channel, something like:

```
        let handle = tokio::spawn(self.extensions.0);
        let guard = tracing::subscriber::set_global_default(self.subscriber);
        future.await;
        self.extensions.0.shutdown(); // sends the shutdown signal, allows hangle to join
        drop(guard);
        handle.await.expect("failed closing the writing task");
```

Then in `async_layer` we have a select!() on rx.recv, so that when the shutdown signal is sent, we can flag as such, then we can have it drain the remaining rx and when it's none, we can shutdown cleanly. 

Then in `AsyncProcessor` for tx.send we can match and if Err() we can have it do an immediate print instead so that if anything is still logging in the BG, the messages are not lost, and go to console instead. 

What do you think?

**Comment by QnnOkabayashi** [comment]
Resolved in #9, closing this issue.
